src/main/java/io/vertx/rabbitmq/impl/QueueConsumerHandler.java

```diff
@@ -4,46 +4,68 @@
 import com.rabbitmq.client.Channel;
 import com.rabbitmq.client.DefaultConsumer;
 import com.rabbitmq.client.Envelope;
+import com.rabbitmq.client.ShutdownSignalException;
 import io.vertx.core.Context;
+import io.vertx.core.Handler;
 import io.vertx.core.Vertx;
 import io.vertx.core.impl.logging.Logger;
 import io.vertx.core.impl.logging.LoggerFactory;
 import io.vertx.rabbitmq.QueueOptions;
 import io.vertx.rabbitmq.RabbitMQConsumer;
 import io.vertx.rabbitmq.RabbitMQMessage;
 
-import static io.vertx.rabbitmq.impl.Utils.put;
-import static io.vertx.rabbitmq.impl.Utils.toJson;
 
 public class QueueConsumerHandler extends DefaultConsumer {
 
   private final RabbitMQConsumerImpl queue;
   private final Context handlerContext;
+  private Handler<ShutdownSignalException> shutdownHandler;
 
-  private static final Logger log = LoggerFactory.getLogger(ConsumerHandler.class);
+  private static final Logger log = LoggerFactory.getLogger(QueueConsumerHandler.class);
 
   QueueConsumerHandler(Vertx vertx, Channel channel, QueueOptions options) {
     super(channel);
     this.handlerContext = vertx.getOrCreateContext();
     this.queue = new RabbitMQConsumerImpl(handlerContext, this, options);
   }
 
+  public void setShutdownHandler(Handler<ShutdownSignalException> shutdownHandler) {
+    this.shutdownHandler = shutdownHandler;
+  }
+
   @Override
   public void handleDelivery(String consumerTag, Envelope envelope, AMQP.BasicProperties properties, byte[] body) {
     RabbitMQMessage msg = new RabbitMQMessageImpl(body, consumerTag, envelope, properties, null);
     this.handlerContext.runOnContext(v -> queue.handleMessage(msg));
   }
 
+  @Override
+  public void handleConsumeOk(String consumerTag) {
+    super.handleConsumeOk(consumerTag);
+    log.info("Consumer tag is now " + consumerTag);
```

Comment:
use debug

Reply:
@vietj Are you OK with me changing it to:
log.debug("consumer has been cancelled unexpectedly: " + consumerTag);
Knowing which consumer is shut down is very useful.

Reply:
yes it is fine